core/src/main/java/org/apache/brooklyn/core/objs/AbstractBrooklynObject.java

```diff
@@ -80,7 +84,7 @@ public AbstractBrooklynObject(Map<?, ?> properties) {
             _legacyConstruction = true;
         }
 
-        catalogItemId = ApiObjectsFactory.get().getCatalogItemIdFromContext();
+        setCatalogItemId(ApiObjectsFactory.get().getCatalogItemIdFromContext());
```

Comment:
Previously `catalogItemId` would be overwritten by any code that knows better. Now most locations should be calling `nestCatalogItemId`. Could this introduce a non-completely-correct inner-most catalog item?